solutions/swb-reference/integration-tests/tests/isolated/projects/assignUserToProject.test.ts

```diff
@@ -0,0 +1,161 @@
+    test('role', async () => {
+      try {
+        await adminSession.resources.projects.project(projectId).assignUserToProject(userId, {});
+      } catch (e) {
+        checkHttpError(
+          e,
+          new HttpError(400, {
+            statusCode: 400,
+            error: 'Bad Request',
+            message: 'role: Required'
+          })
+        );
+      }
+    });
+  });
+
+  test('user does not exist', async () => {
+    const fakeUserId = '00000000-0000-0000-0000-000000000000';
+    const projectId = `${resourceTypeToKey.project.toLowerCase()}-00000000-0000-0000-0000-000000000000`;
+    try {
+      await adminSession.resources.projects
+        .project(projectId)
+        .assignUserToProject(fakeUserId, { role: 'Researcher' });
+    } catch (e) {
+      checkHttpError(
+        e,
+        new HttpError(404, {
+          statusCode: 404,
+          error: 'Not Found',
+          message: `Could not find user ${fakeUserId}`
+        })
+      );
+    }
+  });
+
+  test('admin user cannot be assigned to a project', async () => {
+    let adminUserId: string | undefined = '';
+    const projectId = `${resourceTypeToKey.project.toLowerCase()}-00000000-0000-0000-0000-000000000000`;
+    try {
+      adminUserId = adminSession.getUserId();
+      await adminSession.resources.projects
+        .project(projectId)
+        .assignUserToProject(adminUserId ?? '', { role: 'Researcher' });
+    } catch (e) {
+      checkHttpError(
+        e,
+        new HttpError(400, {
+          statusCode: 400,
+          error: 'Bad Request',
+          message: `IT Admin ${adminUserId} cannot be assigned to the project ${projectId}`
+        })
+      );
+    }
+  });
+
+  describe('user based tests', () => {
+    let userId: string = '';
+    beforeEach(async () => {
+      const response = await adminSession.resources.users.create({
+        firstName: 'Test',
+        lastName: 'User',
+        email: `user-${uuidv4()}@simulator.amazonses.com`
+      });
+
+      const user: User = response.data;
+      userId = user.id;
+    });
+
+    test('project does not exist', async () => {
+      const projectId = `${resourceTypeToKey.project.toLowerCase()}-00000000-0000-0000-0000-000000000000`;
+      try {
+        await adminSession.resources.projects
+          .project(projectId)
+          .assignUserToProject(userId ?? '', { role: 'Researcher' });
+      } catch (e) {
+        checkHttpError(
+          e,
+          new HttpError(404, {
+            statusCode: 404,
+            error: 'Not Found',
+            message: `Could not find project ${projectId}`
+          })
+        );
+      }
+    });
+
+    test('cannot assing user to the same project', async () => {
+      let projectId = '';
+      const projects = await adminSession.resources.projects.get();
+      if (!projects.data.data.length) {
+        console.warn('There are no projects');
+
+        // dummy assertion to make sure that test always passes
```

Comment:
instead of doing this maybe we mark the test as `skip`ped for now until PaaB is done?

Reply:
On a second thought, we do have create project API, let me use that here